pkg/investigations/consoleerrorbudgetburn/consoleerrorbudgetburn.go

```
@@ -569,29 +569,64 @@ func (i *Investigation) checkNLBInstanceSecurityGroups(ctx context.Context, awsC
-	if len(unhealthy) == 0 {
-		notes.AppendSuccess("LB Health: NLB %s — all %d target(s) healthy", lbName, len(targets))
+	var deadPorts []int32
+	for port, stats := range byPort {
+		if stats.healthy == 0 {
+			deadPorts = append(deadPorts, port)
+		}
+	}
+
+	if len(deadPorts) == 0 {
+		// Summarise per port so the note is still informative without being noisy.
+		var portSummaries []string
```

golangci-lint (2.12.2)

[error] Consider preallocating portSummaries with capacity len(byPort)

(prealloc)